### ISSUE TITLE
Can't delete Thesaurus with non ascii terms

### ISSUE THREAD

**Issue by azaroth42** [archesproject/arches#2831]
As a data administrator, I want to delete a thesaurus in RDM as I have imported a new version that should replace the previous version.

_What I would expect:_

When clicking Tools / Delete Scheme, and selecting the thesaurus to be deleted and clicking delete, it should be (you know) deleted.

_What I actually encounter:_

After the spinner spins and the page reloads, the thesaurus is still there.

_Other Information:_

The logging information shows the likely cause:

```
Internal Server Error: /concepts/dd18d680-03ad-4a8c-aa8e-d5bdcd6d9b2c
Traceback (most recent call last):
File "/apps/arches/ENV/lib/python2.7/site-packages/django/core/handlers/base.py", line 149, in get_response
...
 File "/apps/arches/arches/app/views/concept.py", line 233, in concept
    concept.delete(delete_self=delete_self)
   File "/apps/arches/arches/app/models/concept.py", line 247, in delete
     models.Concept.objects.get(pk=key).delete()
...
   File "/apps/arches/ENV/lib/python2.7/site-packages/django/core/files/storage.py", line 286, in delete
     if os.path.exists(name):
   File "/apps/arches/ENV/lib64/python2.7/genericpath.py", line 18, in exists
     os.stat(path)
 UnicodeEncodeError: 'ascii' codec can't encode character u'\\xe4' in position 26: ordinal not in range(128)
```

Clicking the red x to delete individual nodes also fails.

/cc @joshuago78 @workergnome

**Comment by ryan86** [comment]
@azaroth42 I was unable to recreate this with the latest code. Can you please reconfirm this bug?

If the problem persists can you provide more information about the circumstances surrounding the bug and if possible/applicable the rdf file containing the offending character.

In addition, this code from the error: 

```File "/apps/arches/ENV/lib/python2.7/site-packages/django/core/files/storage.py", line 286, in delete
     if os.path.exists(name)```

makes it look like there may be a file associated with this concept. Any details about this would be helpful as well.

**Comment by azaroth42** [comment]
I can't reproduce it now either. Closing until such time as we can reproduce! Apologies.

**Comment by arnaufp** [comment]
so is this solved?

**Comment by azaroth42** [comment]
I'm not sure if it's solved, but I couldn't recreate it with a fresh install.  If you have a particular string that demonstrates the behavior, that would be wonderful! 

Also, did you enter the terms by hand, or import from AAT via SPARQL?

**Comment by arnaufp** [comment]
this happens for example with the Atches > Eligibility Requirement Type  or also with Arches > Component Type > Façade

Also with terms that contain accents that I imported from rdf skos files

![2018-02-19_16h36_09](https://user-images.githubusercontent.com/7857485/36385516-0db2eaf4-1593-11e8-9fa2-69abf2d58db7.png)

**Comment by mradamcox** [comment]
@arnaufp Could you describe your environment and version? e.g. is this a 4.1.0 installation from pypi, or from the current master branch, etc.

**Comment by arnaufp** [comment]
we are using version 4.1. We a fork from the master branch. We are running this on Linux, not Windows.

**Comment by ViktorGetty** [comment]
I stumbled into the same problem. Decided to see if other people had it, and found this thread. We are just testing RDM playing with concepts. So we created manually a Scheme several levels deep by importing from AAT. Then we exported it, and imported into a different Arches box with clean DB. Then by deleting one concept after another I discovered that not all could be deleted. I tried this on 2 different installations, both Ubuntu 16.4, one local, another in the cloud, both running Arches 4.1 on Apache, using Chrome as a browser., with identical result. But the most peculiar thing was that when I ran Arches through Python development server on the same box, I was able to delete all the concepts.

**Comment by mradamcox** [comment]
Wow, great detective work @ViktorGetty. To clarify, when using Apache as the webserver, should _everyone_ add the environment variables you mentioned, or are they only necessary if you are using internationalization? I haven't been able to test this at all, but I'm thinking of what notes should be added to our production setup guide in the docs.

**Comment by ViktorGetty** [comment]
I am not a Unix admin, but this link has a detailed discussion of the issue and solution for Apache2. It contains the other, more specific link, that I have posted on GitHub.

https://itekblog.com/ascii-codec-cant-encode-characters-in-position/

As soon as I modified the ‘envvars’ file and restarted Apache2 service, the problem went away. They also discuss the fact that if running Python dev server, this problem does not come up, which was consistent with my experience.
